bookserver/models.py

```diff
@@ -12,6 +12,8 @@
 #
 # :index:`docs to write`: It is also possible...
 #
+# TODO: What is the database strategy? Who owns/defines/creates which tables? The web2py admin app currently creates and migrates everything. But we're duplicating some tables here. Shouldn't these simply be reflected from the existing database structure, so there's only one "owner" of the table structure (web2py)? But if so, how to correctly interpret web2py booleans?
```

Comment:
I would prefer to have sqlalchemy own the schema.  Especially long term, and even more so because we can create appropriate indices as part of the sqlalchemy table definition.  Which we cannot do with DAL. 

I like your solution for web2py booleans in sqlalchemy, and sqlalchemy has an async/await strategy in place.

**Long term** the admin interface will be ported as well.

Reply:
Having sqlalchemy own the schema long-term makes sense. However, I don't know what a coherent "database migration" strategy looks like. [Migration meaning from web2py to FastAPI/sqlalchemy]. Especially concerning is the thought of Alembic doing migrations on some of the tables while web2py does migrations on other tables. How does this work?

I know sqlalchemy is comfortable with creating Table objects directly from a database (reflection). This makes it easy to use when web2py owns the entire schema. I don't know how/if web2py/DAL can use reflection to infer a table. Assuming it can't, we need to mark some tables as "don't migrate" to web2py -- how is this done? Assuming that's possible, I can see doing this to transfer schema ownership of some tables to sqlalchemy. We'd also need a way to tell Alembic to migrate some tables (the ones owned by sqlalchemy) but not others (owned by web2py).

All if this would be great to capture in the docs, because it will take a while to port everything.

Reply:
I think this is relatively easy?!?  With DAL we can just set a table to NOT do any migrations (migrate=False). which means that DAL will continue to use its own definition of the table without even trying to reflect/and/or update the table in the database.

I *think*?? that for most things this is the right strategy.  But I agree that is a bit confusing until everything is moved over to FastAPI.

Reply:
Thanks, that makes sense! Assuming that Alembic will only migrate the tables defined by this app, then this sounds good. Let's just make sure to capture this approach in the docs.